tests/rptest/tests/prefix_truncate_test.py

```diff
@@ -0,0 +1,201 @@
+# Copyright 2023 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+import random
+from functools import reduce
+from ducktape.mark import parametrize
+from rptest.services.cluster import cluster
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.clients.kcl import KCL
+from rptest.clients.kafka_cli_tools import KafkaCliTools
+from rptest.clients.rpk import RpkTool
+from rptest.services.kafka_cli_consumer import KafkaCliConsumer
+from ducktape.utils.util import wait_until
+from rptest.clients.types import TopicSpec
+from rptest.util import produce_until_segments, wait_for_removal_of_n_segments
+from rptest.services.storage import Segment
+
+
+class PrefixTruncateTest(RedpandaTest):
+    """
+    The purpose of this test is to exercise the delete-records API which
```

Comment:
Other cases to think about:
- DeleteRecords of data which is below max_collectible_offset
- DeleteRecords ahead of hwm
- DeleteRecords below lwm
- DeleteRecords on a partition that doesn't exist
- DeleteRecords on a read replica topic
- DeleteRecords on an empty topic